sirepo/pkcli/job_agent.py

```diff
@@ -55,14 +57,49 @@ def default_command():
     pkdlog('{}', cfg)
     i = tornado.ioloop.IOLoop.current()
     d = _Dispatcher()
-    def s(n, x):
-        return i.add_callback_from_signal(d.terminate)
+    def s(*args):
+        return i.add_callback_from_signal(_terminate, d)
     signal.signal(signal.SIGTERM, s)
     signal.signal(signal.SIGINT, s)
     i.spawn_callback(d.loop)
     i.start()
 
 
+def start_sbatch():
+    def kill_agent(pid_file):
+        if socket.getfqdn() == pid_file.fqdn:
+            os.kill(pid_file.pid, signal.SIGKILL)
+        else:
+            try:
+                subprocess.check_output(
+                    ('ssh', pid_file.fqdn, 'kill', '-KILL', str(pid_file.pid)),
+                    stderr=subprocess.STDOUT,
+                )
+            except subprocess.CalledProcessError as e:
+                pkdp(
+                    'cmd={cmd} returncode={returncode} output={output}',
+                    **vars(e)
+                )
+    try:
+        f = pkio.py_path(_PID_FILE)
+        if f.exists():
+            kill_agent(pkjson.load_any(f))
+    except Exception as e:
+        pkdlog('error={} stack={}', e, pkdexc())
+    pkjson.dump_pretty(
+        PKDict(
+            fqdn=socket.getfqdn(),
+            pid=os.getpid(),
+        ),
+        _PID_FILE,
+        pretty=False,
```

Comment:
irrelevant here since the file is small so less code is better. Perhaps we could have `pkjson.dump_compact`, which would do all the right things.